### PR TITLE
some improvements, promises and chainables

### PR DESCRIPTION
**Pull request by SaulDoesCode**
make the new changes in v1.1 a little more backwards compatible

### CODE REVIEW

File: anime.js [hunk 1]
```diff
@@ -560,33 +570,61 @@
...
       if (!animations.length) engine.pause();
+      return anim;
     }
 
     anim.play = function(params) {
       if (params) anim = mergeObjects(createAnimation(mergeObjects(params, anim.settings)), anim);
       anim.pause();
       anim.running = true;
-      time.start = performance.now();
+      time.start = typeof performance != "undefined" ? performance.now() : +Date.now();
```
Comment (juliangarnier) [inline anime.js]: +Date.now() doesn't return the same values as performance.now() and will breaks the animation.

Comment (SaulDoesCode) [inline anime.js]: oh I see. all the other changes are still valid so I'll just undo that thanks for pointing that out

**Comment by lukebatchelor** [comment]
Hiya @SaulDoesCode, going through the issues and PRs now to see what's happening/where I should be working from. The chainables would be really good for a proof of concept I'm working on. Is it still likely to get in? I noticed a couple of PRs about it but none are merged. 

Other than that, if I was going to take a look at #15 for example, where do you think I should branch from? The es6 branch? Master, v1.1?

Cheers 👍

**Comment by SaulDoesCode** [comment]
Greetings @lukebatchelor , It depends the main repository for anime has a basic promise branch supporting 
```javascript
 animation.once('event').then(func);
 animation.on('update', funct );
``` 
but it still relies on blocking events you cannot have more than one listener as the event's use only one callback per animation per event. The main repo's main branch now has the old 
```javascript 
animation.complete( func );
animation.update ( func ):
```
if you want to fork from the main repos be my guest but the event system is blocking.
if you want to just make methods like animation.play(); animation.complete, .update, .begin and so forth chainable in the main repo then just return the anim variable in the various function's declarations then you can chain them.

My fork [anime-next](https://github.com/SaulDoesCode/anime-next) implements a standard event system and uses it internally to support multiple syntax's for events  and some nice feautures like anime.all for multiple animation event detection. I also implement animation chain's where multiple animations are put into a structure and play one after the other as they complete or finish some event like interloop or so.
note the minified size of anime-next is about 11.6kb more or less so it's not much bigger in comparison but it adds some new functionality.
here are the supported syntax styles in anime-next
```javascript
      // .once returns a promise if there is no callback
  myAnimation.once('begin').then(anim => {
    console.log("Began!"); // Called the animation began.
  });

  myAnimation.once('begin', anim => {
    console.log("Began!"); // Called the animation began.
  });
  // but you can also use
  myAnimation.begin.then(anim => {
    console.log("Began!"); // Called the animation began.
  });
  // or
  myAnimation.begin = anim => {
    console.log("Began!"); // Called the animation began.
  };
  // and for events that get called more than once
 myAnimation.on('begin', anim => {
     console.log("Began!"); // Called the animation began.
 });
```
you can use all of the above for these events below
* update
* complete
* begin      

There are other events that only use the event system's standard syntax

* interloop
* pause
* restart
for these events use
```javascript
 animation.on('event', func ); // -> listener { on , off, once }
 animation.once('event' , func ); // -> listener { on , off, once }
```
So essentially it is up to you I am obviously biased toward my own fork but if you feel you want to start from the main fork and work at your own solution that cool too. but I would recommend you look at the source for [anime-next](https://github.com/SaulDoesCode/anime-next/blob/master/anime.js) to see how the event system works, blocking events is a problem when it comes to cool features like chaining events one after another and using them on interactive websites that use multiple event listeners for animations

**Comment by juliangarnier** [comment]
@lukebatchelor I just created a branch regarding #15, you can PR here https://github.com/juliangarnier/anime/tree/multiple-keys-per-property

**Pull request closed without merging**